test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs

```diff
@@ -400,6 +400,30 @@ public void ParseFilterWithEntityCollectionCount()
                     Source.ShouldBeCollectionNavigationNode(HardCodedTestModel.GetPersonMyFriendsDogsProp());
         }
 
+        [Fact]
+        public void ParseFilterWithEntityCollectionCountWithFilterOption()
+        {
+            var filterQueryNode = ParseFilter("MyFriendsDogs/$count($filter=Color eq 'Brown') gt 1", HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType(), HardCodedTestModel.GetPeopleSet());

```

Comment:
>$filter=Color eq 'Brown' [](http://example.com/codeflow?start=68&length=24)

Do you have a test for $search?  Do you have a test for both $filter and $search?

Reply:
Resolved